spec/models/ability_spec.rb

```diff
@@ -0,0 +1,121 @@
+require 'spec_helper'
+require 'cancan/matchers'
+
+describe Ability do
+  let(:subject) { described_class.new(user) }
+  let(:user) { mock_user }
+  let(:item) { Dor::Item.new(pid: 'x') }
+
+  context 'as an administrator' do
+    let(:user) { mock_user(is_admin?: true)}
+
+    it { should be_able_to(:manage, :everything) }
+  end
+
+  context 'as a manager' do
+    let(:user) { mock_user(is_manager?: true)}
+
+    it { should_not be_able_to(:manage, :everything) }
+    it { should be_able_to(:manage_item, item) }
+    it { should be_able_to(:manage_content, item) }
+    it { should be_able_to(:manage_desc_metadata, item) }
+    it { should be_able_to(:create, Dor::AdminPolicyObject) }
+  end
+
+  context 'as a viewer' do
+    let(:user) { mock_user(is_viewer?: true)}
+
+    it { should_not be_able_to(:manage_item, item) }
+    it { should_not be_able_to(:manage_content, item) }
+    it { should_not be_able_to(:manage_desc_metadata, item) }
+    it { should be_able_to(:view_item, item) }
+    it { should be_able_to(:view_metadata, item) }
+    it { should be_able_to(:view_content, item) }
+  end
+
+  context 'for items without an APO' do
+    it { should_not be_able_to(:manage_item, item) }
+    it { should_not be_able_to(:manage_content, item) }
+    it { should_not be_able_to(:manage_desc_metadata, item) }
+    it { should_not be_able_to(:view_item, item) }
+    it { should_not be_able_to(:view_content, item) }
+    it { should_not be_able_to(:view_metadata, item) }
+  end
+
+  context 'from an APO' do
+    let(:item) { Dor::AdminPolicyObject.new(pid: 'apo') }
+
+    context 'for a managed item' do
+      before { allow(item).to receive(:can_manage_item?).and_return(true) }
+      it { should be_able_to(:manage_item, item) }
+    end
+
+    context 'for a viewable item' do
+      before { allow(item).to receive(:can_view_item?).and_return(true) }
+      it { should be_able_to(:view_item, item) }
+    end
+  end
+
+  context 'with roles assigned by an APO' do
+    let(:apo) { double(Dor::AdminPolicyObject, pid: 'apo')}
+    let(:random_item) { Dor::Item.new(pid: 'x') }
+
+    before do
+      allow(item).to receive(:admin_policy_object).and_return(apo)
+    end
+
+    context 'as a user with a management role for an item' do
+      before do
+        allow(item).to receive(:can_manage_item?).and_return(true)
+      end
+
+      it { should_not be_able_to(:manage_item, random_item) }
+      it { should be_able_to(:manage_item, item) }
+    end
```

Comment:
Yes, perhaps it amounts to more judicious use of `expect` rather than `allow`.  The granularity of these specs is murky in this regard; i.e. the specific code paths the specs execute are not clear.  I don't understand how it could be hitting all the permutations on authorization, starting with Argo global roles (is_admin etc.) and ending with object specific permissions based on APO roles.